source/computer/commands.rst

```diff
@@ -63,54 +62,488 @@ Linux/macOS 下有成百上千个命令，每个命令都有众多选项。这
+    $ cat file1 file2 > target_file
+
+diff
+----
+
+``diff`` 命令的命名来自 **diff**\ erence。该命令可以用来逐行比较文件的异同::
+
+    # 比较 /etc/passwd 和 /etc/passwd- 的异同
+    # 结果显示 /etc/passwd- 多了第 40 行
+    $ diff diff /etc/passwd /etc/passwd-
+    39a40
+    > guest:x:1001:1001::/home/guest:/bin/bash
+
+df
+--
+
+``df`` 命令的命名来自 **d**\ isk **f**\ ree（可使用的磁盘空间）。该命令可以获取硬盘被占用了多少空间，
+目前还剩下多少空间等信息。
+
+::
+
+    $ df -h
+
+du
+--
+
+``du`` 命令的名字来自 **d**\ isk **u**\ sage（磁盘使用情况）。该命令可以查看磁盘的使用空间。
+但与 `df`_ 命令不同的是，\ ``du`` 命令用于查看文件和目录磁盘使用的空间。
+
+查看当前目录所占磁盘空间（也可以在最后指定想查看的目录名）::
+
+    # 默认显示当前目录及其所有子目录所占磁盘空间
+    $ du -h
+
+    # 若只显示当前目录所占磁盘空间，不显示其任何子目录
+    $ du -h --max-depth=0
+
+    # 若只显示当前目录及其一级子目录所占磁盘空间
+    $ du -h --max-depth=1
+
+查看指定文件所占磁盘空间::
+
+    $ du ~/Downloads/TauP-2.4.5.tar.gz
+
 find
 ----
 
+``find`` 命令用来查找指定目录下的子目录和文件，并显示查找到的子目录和文件::
+
+    # 列出 ~/src 目录及其子目录下的所有目录和文件
+    $ find ~/src
+
+    # 查找 ~/src 目录及其子目录下以 .c 结尾的目录和文件
+    $ find ~/src -name "*.c"
+
+    # 同上，但忽略大小
+    $ find ~/src -iname "*.c"
+
 gawk
 ----
 
+``awk`` 命令的命名起源于其三位作者的姓氏首字母。该命令可以选择标准输入、其它命令的输出或文件中
+的特定字段并进行操作。它依次扫描每一行，并读取里面的每一个字段。可以参考此博文
+https://www.ruanyifeng.com/blog/2018/11/awk.html 学习其用法。
+
+``gawk`` 是 GNU 版本的 ``awk`` 命令。Linux 系统下的 ``awk`` 命令一般是指向
+``gawk`` 命令的软链接，可以使用以下命令查看::
+
+    $ ls -l $(which awk)
+    lrwxrwxrwx. 1 root root 4 Aug  4  2018 /usr/bin/awk -> gawk
+
+::
+
+    # 将一个文件的每一行打印到标准输出
+    $ gawk '{print $0}' file
+
+    # 将一个文件的每一行的第 1 个和第 2 个字段打印到标准输出（字段的默认分隔符是空格和制表符）
+    $ gawk '{print $1,$3}' file
+
+    # 同上，但忽略第 1 行和第 2 行，从第 3 行开始
+    $ gawk 'FNR>2 {print $1,$3}' file
+
+    # 将 /etc/passwd 每一行的第 1 个和第 3 个字段打印到标准输出，
+    # 并设置字段分隔符为冒号：
+    $ gawk -F ':' '{print $1,$3}' /etc/passwd
+
+    # 同上，并打印每一行的行号
+    $ gawk -F ':' '{print NR") " $1,$3}' /etc/passwd
+
 grep
 ----
 
+``grep`` 命令的命名来自 **g**\ lobally search a **re**\ gular expression and **p**\rint
+（以正则表达式进行全局搜索并打印）。它是一个强大的文本搜索工具，可以搜索文件中与指定模式匹配的行并输出::
+
+    # 搜索并打印 /etc/passwd 中含有 root 的行
+    $ grep root /etc/passwd
+
+    # 同上，并显示行号
+    $ grep -n root /etc/passwd
+
+    # 搜索并打印 /etc/passwd 中不含有 root 的行
+    $ grep -v root /etc/passwd
+
 head
 ----
 
+``head`` 命令用于打印文件开头部分的内容::
+
+    # 打印 /etc/passwd 的前 10 行
+    $ head /etc/passwd
+
+    # 打印 /etc/passwd 的前 5 行
+    $ head -n 5 /etc/passwd
+
 less
 ----
 
+``less`` 命令可以用来浏览文件内容，比 ``more`` 命令功能更强大。因此，一个常用的笑话记忆方式是
+“less is more”。该命令允许用户向前或向后浏览文件：按 :kbd:`PageUp` 键向上翻页，
+用 :kbd:`PageDown` 键向下翻页，按 :kbd:`Enter` 键或向下方向键则向下移动，
+用向上方向键则向上移动。按 :kbd:`Q` 键退出浏览。
+
+::
+
+    # 浏览 /etc/passwd
+    $ less /etc/passwd
+
+    # 同上，并显示行号
+    $ less -N /etc/passwd
+
 locate
 ------
 
+``locate`` 命令可以用于查找目录和文件。该命令比 `find`_ 命令快得多，原因在于它不搜索具体目录，
+而是搜索含有本地所有文件信息的数据库::
+
+    # 搜索名称含 passwd 的目录或文件
+    $ locate passwd
+
+    # 搜索名称含 /etc/sh 的目录或文件（包括了 /etc 目录下所有以 sh 开头的目录或文件）
+    $ locate /etc/sh
+
+    # 搜索名称含 ~/Des 的目录或文件（包括了 ~ 家目录下，所有以 Des 开头的目录或文件）
+    $ locate ~/Des
+
+    # 同上，但忽略大小写
+    $ locate -i ~/des
+
 rsync
 -----
 
```

Comment:
Let's remove this command here? I want to put it in file backup.

Reply:
`rsync` is also used in #103.

Reply:
I see.